plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -597,6 +599,17 @@ class Pub(
                         logger.warn { "No version information found for package $rawName." }
                     }
 
+                    val hostUrl = pkgInfoFromLockFile["description"]["url"].textValueOrEmpty()
```

Comment:
Line 588 is written as if `pkgInfoFromLockFile["description"]` could be null. So, I believe line 602 needs to handle this as well. I'm not 100% certain, but would `pkgInfoFromLockFile["description"]["url"]` throw an NPE if there is no value for key `"description"`?

Reply:
Which line do you mena, line 588 reads just `}`?

Reply:
588 in old file, 590 in new file.

Reply:
`pkgInfoFromLockfile["description"]` is never null (at least in our test data), but it can be either a node or a text. If it's a text `pkgInfoFromLockfile["description"]["url"]` will return `null` so there is no need for special handling.

Reply:
But then the `textValueOrEmpty()` should be changed to just `textValue()` shouldn't it?

Reply:
> If there is really guarantee that it's never null, then the `textValueOrEmpty()` should be changed to just `textValue()` shouldn't it?

To me, ideally any such assumptions are not subtly expressed via `textValueOrEmpty()` vs `textValue()`, but via explicit `checkNotNull()` statements.

Reply:
`pkgInfoFromLockfile["description"]` cannot be null but `pkgInfoFromLockfile["description"]["url"]` can be, so `textValueOrEmpty()` is required.